types/src/test_utils.rs

```diff
@@ -0,0 +1,449 @@
+use ed25519_consensus::SigningKey;
+use tendermint::{
+    block::{
+        header::{Header, Version},
+        Commit, CommitSig,
+    },
+    chain,
+    public_key::PublicKey,
+    Signature, Time,
+};
+
+use crate::block::{CommitExt, GENESIS_HEIGHT};
+use crate::consts::version;
+use crate::hash::{Hash, HashExt};
+use crate::nmt::{NamespacedHash, NamespacedHashExt};
+use crate::{DataAvailabilityHeader, ExtendedHeader, ValidatorSet};
+
+#[derive(Debug, Clone)]
+pub struct ExtendedHeaderGenerator {
+    chain_id: chain::Id,
+    key: SigningKey,
+    current_header: Option<ExtendedHeader>,
+}
+
+/// [`ExtendedHeader`] generator for testing purposes.
+///
+/// **WARNING: ALL METHODS PANIC! DO NOT USE IT IN PRODUCTION!**
+impl ExtendedHeaderGenerator {
+    /// Creates new `ExtendedHeaderGenerator`.
+    pub fn new() -> ExtendedHeaderGenerator {
+        let chain_id: chain::Id = "private".try_into().unwrap();
+
+        let key = SigningKey::new(rand::thread_rng());
+
+        ExtendedHeaderGenerator {
+            chain_id,
+            key,
+            current_header: None,
+        }
+    }
+
+    /// Creates new `ExtendedHeaderGenerator` and skip an amount of headers.
+    ///
+    /// This is equivalent to:
+    ///
+    /// ```ignore
+    /// let mut gen = ExtendedHeaderGenerator::new();
+    /// gen.skip(amount);
+    /// ```
+    pub fn new_skipped(amount: u64) -> ExtendedHeaderGenerator {
```

Comment:
I'd rather prefer something like `new_from_height(height: u64)` that generates the first header of `height` height. I found this `new_skipped` a bit weird to understand

Reply:
> generates the first header of `height` height.

I don't want to return a `(ExtendedHeaderGenerator, ExtendedHeader)`, `new_from_height` as a name is ok with me.

Reply:
done